src/js/content_scripts/input-mapper-utils.js

```diff
@@ -0,0 +1,182 @@
+     * @param {Integer} value - The value of the gamepad input.
+     *
+     */
+    gamepad.inputMapperUtils.scrollRight = function (that, value) {
+        /**
+         * Stop scrolling for the previous input value. Also stop scrolling if the input
+         * source (analog/button) is at rest.
+         */
+        clearInterval(that.options.intervalRecord.rightScroll);
+
+        /**
+         * Scroll the webpage towards the right only if the input value is more than the
+         * specified value.
+         */
+        if (value > 0.20) {
+            // Scroll to the right according to the new input value.
+            that.options.intervalRecord.rightScroll = setInterval(function () {
+                var xOffset = $(window).scrollLeft();
+                $(window).scrollLeft(xOffset + value * 50);
+            }, that.options.frequency);
+        }
+    };
+
+    /**
+     *
+     * Scroll vertically across the webpage.
+     *
+     * @param {Object} that - The inputMapper component.
+     * @param {Integer} value - The value of the gamepad input.
+     *
+     */
+    gamepad.inputMapperUtils.scrollVertically = function (that, value) {
+        /**
+         * Scroll the webpage vertically only if the input's absolute value is more than
+         * the specified value. Otherwise, stop scrolling across the webpage.
+         */
+        if (value > 0) {
+            that.scrollDown(value);
+        }
+        else if (value < 0) {
+            that.scrollUp(-1 * value);
+        }
+        else {
+            clearInterval(that.options.intervalRecord.upwardScroll);
+            clearInterval(that.options.intervalRecord.downwardScroll);
+        }
+    };
+
+    /**
+     *
+     * Scroll the webpage in upward direction.
+     *
+     * @param {Object} that - The inputMapper component.
+     * @param {Integer} value - The value of the gamepad input.
+     *
+     */
+    gamepad.inputMapperUtils.scrollUp = function (that, value) {
+        /**
+         * Stop scrolling for the previous input value. Also stop scrolling if the input
+         * source (analog/button) is at rest.
+         */
+        clearInterval(that.options.intervalRecord.upwardScroll);
+
+        /**
+         * Scroll the webpage upward only if the input value is more than the specified
+         * value.
+         */
+        if (value > 0.20) {
+            // Scroll upward according to the new input value.
+            that.options.intervalRecord.upwardScroll = setInterval(function () {
+                var yOffset = $(window).scrollTop();
+                $(window).scrollTop(yOffset - value * 50);
+            }, that.options.frequency);
+        }
+    };
+
+    /**
+     *
+     * Scroll the webpage in downward direction.
+     *
+     * @param {Object} that - The inputMapper component.
+     * @param {Integer} value - The value of the gamepad input.
+     *
+     */
+    gamepad.inputMapperUtils.scrollDown = function (that, value) {
```

Comment:
I think you should have a single function with an "axis" argument rather than having these two very similar functions.